lib/Common/Memory/Allocator.h

```diff
@@ -227,6 +227,7 @@ struct AllocatorInfo
 template <typename TAllocator>
 struct ForceNonLeafAllocator
 {
+    static const bool FakeZeroLengthArray = true;
```

Comment:
see https://github.com/Microsoft/ChakraCore/pull/2939/commits/5224e0583a1da50c529e1bbfbf67d81c04354a39

while Heap allocation zero length array, we allocated the track data, but we didn't delete the track data in DeleteArray, causes memory leak in chk build.

with changing DeleteArray to use the FakeZeroLengthArray, every allocator need to provide such definition.